### PR TITLE
Update filter functionality for Module & Faculty

### LINKED ISSUE #8: As a slow reader, I can filter through contact entry by their module
... so that I would not waste time in finding a specific contact/s and access their info easily

### LINKED ISSUE #13: As forgetful user, I can filter the professors by their faculty or the course they teach
... so that I can find professors based on my needs but with no special preference

### LINKED ISSUE #49: Implement filter functionality for contacts
Filter by
 - [ ] Module
 - [ ] Faculty
 - [x] Tags

### PR DESCRIPTION
**Pull request by iynixil**
Closes #49

### CODE REVIEW

File: src/main/java/staffconnect/model/person/PersonHasFacultyPredicate.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public PersonHasFacultyPredicate(Faculty faculty) {
+        this.faculty = faculty;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (faculty == null) {
+            return true;
+        }
+        return person.getFaculty().value.getFacultyName().equalsIgnoreCase(faculty.value.getFacultyName());
```
Comment (whitesnowx) [inline src/main/java/staffconnect/model/person/PersonHasFacultyPredicate.java:22]: Can consider using `person.getFaculty().toString().equalsIgnoreCase(faculty.toString());` instead for consistency

File: src/main/java/staffconnect/model/person/PersonHasModulePredicate.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public PersonHasModulePredicate(Module module) {
+        this.module = module;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (module == null) {
+            return true;
+        }
+        return person.getModule().value.equalsIgnoreCase(module.value);
```
Comment (whitesnowx) [inline src/main/java/staffconnect/model/person/PersonHasModulePredicate.java:22]: similar for here: `person.getModule().toString().equalsIgnoreCase(module.toString());`

**Review by whitesnowx — COMMENTED** [review]
can consider my suggestions for the next iterations 
everything else LTGM 🍾 ✨

**Review by whitesnowx — APPROVED** [review]

File: src/main/java/staffconnect/model/person/PersonHasFacultyPredicate.java (hunk 1 above)
Comment (iynixil) [reply src/main/java/staffconnect/model/person/PersonHasFacultyPredicate.java:22]: will take note!

**Pull request merged**